homeassistant/components/sensoterra/sensor.py

```diff
@@ -0,0 +1,139 @@
+from .models import ProbeSensorType
+
+SENSORS: dict[ProbeSensorType, SensorEntityDescription] = {
+    ProbeSensorType.MOISTURE: SensorEntityDescription(
+        key=ProbeSensorType.MOISTURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.MOISTURE,
+        native_unit_of_measurement=PERCENTAGE,
+        translation_key="soil_moisture_at_cm",
+    ),
+    ProbeSensorType.SI: SensorEntityDescription(
+        key=ProbeSensorType.SI,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        translation_key="si_at_cm",
+    ),
+    ProbeSensorType.TEMPERATURE: SensorEntityDescription(
+        key=ProbeSensorType.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+    ),
+    ProbeSensorType.BATTERY: SensorEntityDescription(
+        key=ProbeSensorType.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    ProbeSensorType.RSSI: SensorEntityDescription(
+        key=ProbeSensorType.RSSI,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SensoterraConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up Sensoterra sensor."""
+
+    coordinator = entry.runtime_data
+
+    @callback
+    def _async_add_devices(st_sensors: list[SensoterraSensor]) -> None:
+        async_add_devices(
+            SensoterraEntity(coordinator, st_sensor.probe, st_sensor.sensor)
+            for st_sensor in st_sensors
+            if st_sensor.sensor.type.lower() in SENSORS
+        )
+
+    coordinator.add_devices_callback = _async_add_devices
+
+    _async_add_devices(coordinator.data)
+
+
+class SensoterraEntity(CoordinatorEntity[SensoterraCoordinator], SensorEntity):
+    """Sensoterra sensor like a soil moisture or temperature sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: SensoterraCoordinator,
+        probe: Probe,
+        sensor: Sensor,
+    ) -> None:
+        """Initialize entity."""
+        super().__init__(coordinator, context=sensor.id)
+
+        self._attr_unique_id = sensor.id
+        self._attr_translation_placeholders = {
+            "depth": "?" if sensor.depth is None else str(sensor.depth)
+        }
```

Comment:
I would rather see a separate sensor telling the depth

Reply:
Depth is not something you measure, it's a physical propery of the soul moisture sensor: the length of the pins hammered into the soil. Some probes even have six soil moisture sensors at 6 different fixed depths. For the user it's important to know which sensor at which depth the soil moisture sensor is reporting. That's why depth is part of the sensor name.